File: packages/api/src/endpoints/treasury.ts
import type { TreasuryDonationStatsResponse } from "@/types/treasuryTypes";

import { handleFetch } from "@/lib/handleFetch";

/**
 * Fetches statistics related to treasury donations.
 *
 * @returns {Promise<TreasuryDonationStatsResponse>} A promise that resolves to treasury donation statistics.
 *
 * @throws Will throw an error if the fetch request fails.
 */
export const getTreasuryDonationStats = async () => {
  const url = "/analytics/treasury";

  return handleFetch<TreasuryDonationStatsResponse>(url);
};

File: packages/cexplorer-api-rs/src/endpoints/treasury.rs
use crate::client::fetch;
use crate::error::CexplorerError;
use crate::types::treasury_types::*;

pub async fn get_treasury_donation_stats() -> Result<TreasuryDonationStatsResponse, CexplorerError> {
    let endpoint = "/analytics/treasury";
    fetch::<TreasuryDonationStatsResponse>(endpoint).await
}
